File: TiresShopApp/src/main/java/org/tireshop/tiresshopapp/entity/Address.java
package org.tireshop.tiresshopapp.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "addresses")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Address {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;

  private String street;
  private String houseNumber;
  private String apartmentNumber;
  private String postalCode;
  private String city;

  @Enumerated(EnumType.STRING)
  private AddressType type;

  @ManyToOne(fetch = FetchType.LAZY)
  @JoinColumn(name = "user_id")
  private User user;
}

File: TireShopFrontend/src/types/Address.ts
export type AddressType = 'BILLING' | 'SHIPPING' | 'RESIDENTIAL';

export interface Address {
    id: number;
    street: string;
    houseNumber: string;
    apartmentNumber: string;
    postalCode: string;
    city: string;
    type: AddressType;
}